### PR TITLE
New version: Thales.SafeNetAuthenticationClient version 10.9.3283.0

### LINKED ISSUE #173741: [Update Request]: Thales.SafeNetAuthenticationClient  version 10.9.3283.0   (arm64 + x86 + x64) = CVE-2023-7016 patch
### What type of update are you requesting?

A new version of an existing package

### Current Package Identifier

Thales.SafeNetAuthenticationClient

### Package Version

10.9.3283.0

### Please describe the changes you would like to see

General download info:  https://knowledge.digicert.com/general-information/how-to-download-safenet-authentication-client

**ZIP file, with installers for all 3 platforms (arm64 + x86 + x64) - multiple MSI installers in a single ZIP file:
https://www.digicert.com/StaticFiles/Windows_SAC_10.9_GA.zip**

`RelativeFilePath` locations inside the zip file: 
```
RelativeFilePath: SAC 10.9 GA\Msi\arm64\SafeNetAuthenticationClient-arm64-10.9.msi
RelativeFilePath: SAC 10.9 GA\Msi\x32\SafeNetAuthenticationClient-x32-10.9.msi
RelativeFilePath: SAC 10.9 GA\Msi\x64\SafeNetAuthenticationClient-x64-10.9.msi
```
(edit note: fixed missing top level folder in that file path there. /edit)

Thales / Gemalto release announcement:
https://data-protection-updates.gemalto.com/2024/04/19/safenet-authentication-client-sac-10-9-for-windows-and-safenet-minidriver-10-9-release-announcement/

CVE-2023-7016 Detail:
https://nvd.nist.gov/vuln/detail/CVE-2023-7016
A flaw in Thales SafeNet Authentication Client prior to 10.8 R10 on Windows allows an attacker to execute code at a SYSTEM level via local access.

### PR DESCRIPTION
**Pull request by Dragon1573**
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #173741 

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/173818)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> No errors to post.

(Automated response - build 895.)

**Comment by R-Adrian** [comment]
question: does the slash direction matter for syntax validation in `RelativeFilePath`?
i see that this PR uses forward slashes while most other PRs i've seen use back slashes

e.g.
`RelativeFilePath: Msi/x64/SafeNetAuthenticationClient-x64-10.9.msi`
vs
`RelativeFilePath: Msi\x64\SafeNetAuthenticationClient-x64-10.9.msi`

**Review by ItzLevvie — APPROVED** [review]

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335141
> 2024-09-17 03:19:00.701 [CLI ] Unable to locate nested installer at: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Thales.SafeNetAuthenticationClient.10.9.3283.0\extracted\Msi/x64/SafeNetAuthenticationClient-x64-10.9.msi
> 2024-09-17 03:18:52.553 [CLI ] Unable to locate nested installer at: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Thales.SafeNetAuthenticationClient.10.9.3283.0\extracted\Msi/x32/SafeNetAuthenticationClient-x32-10.9.msi

Automated error analysis suggests -1978335141 may mean Unable to locate nested installer. Does the `RelativeFilePath` have the previous version number in it?

(Automated response - build 895.)

**Comment by stephengillie** [comment]
> question: does the slash direction matter for syntax validation in `RelativeFilePath`? i see that this PR uses forward slashes while most other PRs i've seen use back slashes
> 
> e.g. `RelativeFilePath: Msi/x64/SafeNetAuthenticationClient-x64-10.9.msi` vs `RelativeFilePath: Msi\x64\SafeNetAuthenticationClient-x64-10.9.msi`
> 
> update: looking at the ZIP format APPNOTE.TXT specification it looks like forward slashes is actually the correct format. quote:
> 
> ```
>    4.4.17 file name: (Variable)
> 
>        4.4.17.1 The name of the file, with optional relative path.
>        The path stored MUST NOT contain a drive or
>        device letter, or a leading slash.  All slashes
>        MUST be forward slashes '/' as opposed to
>        backwards slashes '\' for compatibility with Amiga
>        and UNIX file systems etc.  If input came from standard
>        input, there is no file name field.  
> ```

This is something that we're not currently checking, but probably should be. We usually don't notice it, I believe, because the string is likely used on a Windows system after replacing the slashes for backslashes,

**Comment by R-Adrian** [comment]
> > update: looking at the ZIP format APPNOTE.TXT specification it looks like forward slashes is actually the correct format. quote:
> > ```
> >    4.4.17 file name: (Variable)
> > 
> >        4.4.17.1 The name of the file, with optional relative path.
> >        The path stored MUST NOT contain a drive or
> >        device letter, or a leading slash.  All slashes
> >        MUST be forward slashes '/' as opposed to
> >        backwards slashes '\' for compatibility with Amiga
> >        and UNIX file systems etc.  If input came from standard
> >        input, there is no file name field.  
> > ```
> 
> This is something that we're not currently checking, but probably should be. We usually don't notice it, I believe, because the string is likely used on a Windows system after replacing the slashes for backslashes,


Considering the nature of the validation error message i think that the validation servers ARE actually checking it but in a windows-specific way and they are doing it EXACTLY the WRONG way, that the ZIP standard says you MUST NOT do... the validation servers are interpreting the forward slash as part of the file name and not as part of the directory path.

See the way that the error message looks: the first part is back slashes... the second part it forward slashes.

`2024-09-17 03:18:52.553 [CLI ] Unable to locate nested installer at: C:\Users\VALIDA1\AppData\Local\Temp\WinGet\Thales.SafeNetAuthenticationClient.10.9.3283.0\extracted\Msi/x32/SafeNetAuthenticationClient-x32-10.9.msi
`

**Review by stephengillie — APPROVED** [review]

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @Dragon1573,

It appears that a folder is missing from the `RelativeFilePath`:
image

File: manifests/t/Thales/SafeNetAuthenticationClient/10.9.3283.0/Thales.SafeNetAuthenticationClient.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- Architecture: x64
+  InstallerUrl: https://www.digicert.com/StaticFiles/Windows_SAC_10.9_GA.zip
+  InstallerSha256: 3B963D24C0B145775B8ECD950D8568DC44EBA7A872792C009D0287DCD7AAC034
+  NestedInstallerFiles:
+  - RelativeFilePath: Msi/x64/SafeNetAuthenticationClient-x64-10.9.msi
+- Architecture: arm64
+  InstallerUrl: https://www.digicert.com/StaticFiles/Windows_SAC_10.9_GA.zip
+  InstallerSha256: 3B963D24C0B145775B8ECD950D8568DC44EBA7A872792C009D0287DCD7AAC034
+  NestedInstallerFiles:
+  - RelativeFilePath: Msi/arm64/SafeNetAuthenticationClient-arm64-10.9.msi
```
Comment (stephengillie) [inline manifests/t/Thales/SafeNetAuthenticationClient/10.9.3283.0/Thales.SafeNetAuthenticationClient.installer.yaml:25]: I thought I had clicked "Start a review" to add this to the above review, but must have forgotten.
```suggestion
  - RelativeFilePath: SAC 10.9 GA/Msi/x32/SafeNetAuthenticationClient-x32-10.9.msi
- Architecture: x64
  InstallerUrl: https://www.digicert.com/StaticFiles/Windows_SAC_10.9_GA.zip
  InstallerSha256: 3B963D24C0B145775B8ECD950D8568DC44EBA7A872792C009D0287DCD7AAC034
  NestedInstallerFiles:
  - RelativeFilePath: SAC 10.9 GA/Msi/x64/SafeNetAuthenticationClient-x64-10.9.msi
- Architecture: arm64
  InstallerUrl: https://www.digicert.com/StaticFiles/Windows_SAC_10.9_GA.zip
  InstallerSha256: 3B963D24C0B145775B8ECD950D8568DC44EBA7A872792C009D0287DCD7AAC034
  NestedInstallerFiles:
  - RelativeFilePath: SAC 10.9 GA/Msi/arm64/SafeNetAuthenticationClient-arm64-10.9.msi
```

**Review by ItzLevvie — APPROVED** [review]

**Review by stephengillie — APPROVED** [review]

**Pull request merged**